subprojects/packagefiles/libtommath/meson.build

```diff
@@ -0,0 +1,386 @@
+  'bn_mp_sqrmod.c',
+  'bn_mp_sqrt.c',
+  'bn_mp_sqrtmod_prime.c',
+  'bn_mp_sub.c',
+  'bn_mp_sub_d.c',
+  'bn_mp_submod.c',
+  'bn_mp_to_radix.c',
+  'bn_mp_to_sbin.c',
+  'bn_mp_to_ubin.c',
+  'bn_mp_ubin_size.c',
+  'bn_mp_unpack.c',
+  'bn_mp_xor.c',
+  'bn_mp_zero.c',
+  'bn_prime_tab.c',
+  'bn_s_mp_add.c',
+  'bn_s_mp_balance_mul.c',
+  'bn_s_mp_div_3.c',
+  'bn_s_mp_exptmod.c',
+  'bn_s_mp_exptmod_fast.c',
+  'bn_s_mp_get_bit.c',
+  'bn_s_mp_invmod_fast.c',
+  'bn_s_mp_invmod_slow.c',
+  'bn_s_mp_karatsuba_mul.c',
+  'bn_s_mp_karatsuba_sqr.c',
+  'bn_s_mp_log.c',
+  'bn_s_mp_log_2expt.c',
+  'bn_s_mp_log_d.c',
+  'bn_s_mp_montgomery_reduce_fast.c',
+  'bn_s_mp_mul_digs.c',
+  'bn_s_mp_mul_digs_fast.c',
+  'bn_s_mp_mul_high_digs.c',
+  'bn_s_mp_mul_high_digs_fast.c',
+  'bn_s_mp_prime_is_divisible.c',
+  'bn_s_mp_rand_jenkins.c',
+  'bn_s_mp_rand_platform.c',
+  'bn_s_mp_reverse.c',
+  'bn_s_mp_sqr.c',
+  'bn_s_mp_sqr_fast.c',
+  'bn_s_mp_sub.c',
+  'bn_s_mp_toom_mul.c',
+  'bn_s_mp_toom_sqr.c',
+)
+
+inc = include_directories('.')
+
+# Add extra optimization flags to match Makefile and CMake builds
+# Note: Meson already adds -O3 when optimization=3 (from buildtype=release)
+# We only need to add the flags Meson doesn't add by default
+if not get_option('debug')
+  add_project_arguments(
+    cc.get_supported_arguments('-funroll-loops', '-fomit-frame-pointer'),
+    language: 'c',
+  )
+endif
+
+# Build both static and shared libraries by default
+# Users can override with -Ddefault_library=static or -Ddefault_library=shared
+libtommath = library(
+  'tommath',
+  sources,
+  install: true,
+  include_directories: inc,
+  version: meson.project_version(),
+  link_args: cc.get_id() in ['msvc', 'clang-cl'] ? [
+    # MSVC by default exports no symbols, leading to no .LIB file being generated
+    '/DEF:@0@'.format(meson.current_source_dir() / 'libtommath.def'),
+  ] : [],
+  c_args: cc.get_id() in ['msvc', 'clang-cl'] ? [
+    # MSVC defaults to IEEE754 by default since VS 2010, but doesn't set the macro.
+    # Can't use add_project_arguments here because that enables tests that contain
+    # literal division by zero, which breaks compilation.
+    '/fp:precise',
+    '-D__STDC_IEC_559__',
+  ] : [],
+)
+
+# Install headers
+install_headers(files('tommath.h'))
+
+# Generate pkg-config file
+pkg = import('pkgconfig')
+pkg.generate(
+  libtommath,
+  name: meson.project_name(),
+  description: 'public domain library for manipulating large integer numbers',
+  url: 'https://www.libtom.net/LibTomMath/',
+)
+
+# Create a dependency object for use by other projects
+# This makes the library usable as a Meson subproject
+libtommath_dep = declare_dependency(
+  link_with: libtommath,
+  include_directories: inc,
+)
+
+# Make the dependency available to parent projects
+meson.override_dependency(meson.project_name(), libtommath_dep)
+
+# Build and run unit tests if requested
+if get_option('tests')
+  tommath_test_exe = executable(
```

Comment:
Nit: there's some value to `build_by_default: false` for these (it works correctly on newer Meson at least) because it's possible to build with tests enabled but never actually run tests.